main.cpp

```diff
@@ -92,6 +92,72 @@ good to go!
  */
 
 #include <iostream>
+
+
+class FloatType
+{
+    public:
+        float add(float lhs, float rhs);
+        float subtract(float lhs, float rhs);
+        float multiply(float lhs, float rhs );  
+        float divide(float lhs, float rhs );
+};
+
+float FloatType::add(float lhs, float rhs) { return lhs + rhs; }
+float FloatType::subtract(float lhs, float rhs) { return lhs - rhs; }
+float FloatType::multiply(float lhs, float rhs) { return lhs * rhs; }
+float FloatType::divide(float lhs, float rhs)
+{
+    if (0.0f == rhs) {
+        std::cout << std::endl;
+        std::cout << "warning, floating point division by zero returns 'inf' !" << std::endl;
+    }
+    return lhs / rhs;
+}
+
+class DoubleType
```

Comment:
use `struct` not `class`, so you don't have to type `public` all the time
- [ ] revise